plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Payments.php

```diff
@@ -219,6 +224,33 @@ private function is_woopayments_onboarded(): bool {
 		return false;
 	}
 
+	/**
+	 * Check if WooPayments has a live account onboarding in progress.
+	 *
+	 * @return bool
+	 */
+	private function has_woopayments_live_account_in_progress() {
+		if ( $this->is_woopayments_onboarded() ) {
+			return false;
+		}
+
+		$woopayments_provider = $this->get_woopayments_provider();
+		// We should have the WooPayments provider, but if not, return false.
+		if ( ! $woopayments_provider ) {
+			return false;
+		}
+
+		// Check the provider's state to determine if a live account onboarding is in progress.
+		if ( ! empty( $woopayments_provider['onboarding']['state']['started'] ) &&
+			empty( $woopayments_provider['onboarding']['state']['completed'] ) &&
+			empty( $woopayments_provider['onboarding']['state']['test_drive_account'] )
```

Comment:
I would pull out the test_drive_account condition and replace it with `$this->has_woopayments_test_account()`, again with a "bail early" approach.

Reply:
Great call, replaced with `$this->has_woopayments_test_account()` and pulled out to bail early. Thanks!